File: PSK2025.Models/Entities/Project.cs
using PSK2025.Models.Enums;

namespace PSK2025.Models.Entities;

public class Project
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;

    // Required foreign key to Identity user
    public string OwnerId { get; set; } = string.Empty;

    // Required navigation property
    public User Owner { get; set; } = null!;

    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public string? Description { get; set; }

    public ProjectStatus Status { get; set; }

    public ICollection<Task> Tasks { get; set; } = new List<Task>();
    public ICollection<UserProject> UserProjects { get; set; } = new List<UserProject>();
}

File: PSK2025.Web/app/types/project.ts
export interface Project {
  id: string;
  name: string;
  owner: {
    id: string;
    firstName: string;
    lastName: string;
  }
  startDate: string | null;
  endDate: string | null;
  description: string | null;
  status: ProjectStatus;
}

export interface NestedProject {
  project: Project;
}

export enum ProjectStatus {
  PLANNED,
  ACTIVE,
  COMPLETED,
  ARCHIVED
}

export interface CreateProjectRequest {
  name: string;
  startDate: string | null;
  endDate: string | null;
  description: string | null;
  status: ProjectStatus;
}

export interface UpdateProjectRequest {
  name: string;
  startDate: string | null;
  endDate: string | null;
  description: string | null;
  status: ProjectStatus;
}
